### DISCUSSION TITLE
Unable to SSH login at all after disabling root login via SSH

### DISCUSSION THREAD

**Discussion by centipedesquad** [MichaIng/DietPi#7203] [Q&A] [answered]
I was following the instructions here to disable root password access. I did this:
`/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins root
dietpi-software reinstall 104 # reinstall Dropbear`

However, since I reinstalled Dropbear and then restarted my RPi 5, I am unable to ssh into my user (not root user) using the password either. I am getting an `ssh: connect to host xxx: Operation timed out` error when trying to log in to either root or the other user.

Since by default, I don't have the location `/etc/ssh/sshd_config.d/zz-disable-root-login.conf` and I am getting this error:
`bash: /etc/ssh/sshd_config.d/zz-disable-root-login.conf: No such file or directory`
(because I assume it is specific to openssh and not Dropbear)
I hadn't performed this step.

Any suggestions as to what might be the problem, and what could I do to resolve it?

Unfortunately, just before this I have also installed and run fail2ban, so I am not sure which of the changes is causing the error.

Apologies. I am new to all this, so I might have made a rather elementary error.

Thanks!

**Comment by Joulinar** [comment]
Easiest way would be to get local consol access by connecting screen and keyboard.

**Reply by centipedesquad** [reply]
Thank you @Joulinar could you also suggest where and what I should look for to correct this error?

**Reply by Joulinar** [reply]
are you able to login locally now? If yes, you can run following to revert the settings

```
/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins 0
```

**Reply by MichaIng** [reply]
If is still does not work, please check
```sh
journalctl -u dropbear
```

It is not needed to reinstall Dropbear after the change, restarting its service is sufficient. The config is at `/etc/default/dropbear`. Can you run the command again and paste the content of this config file?
```sh
/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins root
cat /etc/default/dropbear
```

... actually something to add to our docs: https://dietpi.com/docs/software/ssh/

**Reply by centipedesquad** [reply]
Thanks! 
Not yet, I hadn't realised that a USB-C monitor can't be connected to RBP while simultaneously powering it. So I have ordered a micro-HDMI to HDMI adapter and will try connecting it to my TV. Will post as soon as I am able to do so.

**Reply by MichaIng** [reply]
A UART adapter for serial console access would be an alternative.

Also, if you have another Linux system where you can attack the drive/SD card to, you could manually check and in case edit the Dropbear config from there.

**Reply by centipedesquad** [reply]
> If is still does not work, please check
> 
> ```shell
> journalctl -u dropbear
> ```
> 
> It is not needed to reinstall Dropbear after the change, restarting its service is sufficient. The config is at `/etc/default/dropbear`. Can you run the command again and paste the content of this config file?
> 
> ```shell
> /boot/dietpi/func/dietpi-set_software disable_ssh_password_logins root
> cat /etc/default/dropbear
> ```
> 
> ... actually something to add to our docs: https://dietpi.com/docs/software/ssh/
Here is what has happened:

1. I ran `/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins 0` after connecting the RPi to a monitor and keyboard. Then after restarting the Pi, then I was _able_ to login using ssh to any user.

2. Then, I ran `/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins root` and then restarted the Pi. I was unable to login through ssh again. After that, I was able to login through ssh the moment I connected an external monitor, but without changing anything locally. It happened rather accidentally, I ran the ssh command turned to connect the external monitor just to see what it was displaying. When I turned back again and put in the password it logged in. Now it is true that one might say that I might have input the password incorrectly but since I was trying to eliminate any such possibility I was pasting my password into the iterm terminal anyways :)
Since then I have been able to ssh login into all users except the root as I would expect it to work. Here are the outputs in any case

- `journalctl -u dropbear` returns:

> xxxxxx@DietPi:~$ sudo journalctl -u dropbear
> Sep 06 20:48:58 DietPi systemd[1]: Started dropbear.service - Lightweight SSH server.
> Sep 06 20:48:58 DietPi dropbear[527]: [527] Sep 06 20:48:58 Failed loading /etc/dropbear/dropbear_dss_host_key
> Sep 06 20:48:58 DietPi dropbear[527]: [527] Sep 06 20:48:58 Not backgrounding
> Sep 06 20:48:59 DietPi dropbear[578]: [578] Sep 06 20:48:59 Child connection from 192.168.1.193:61455
> Sep 06 20:48:59 DietPi dropbear[578]: [578] Sep 06 20:48:59 Failed loading /etc/dropbear/dropbear_dss_host_key
> Sep 06 20:49:43 DietPi dropbear[578]: [578] Sep 06 20:49:43 Exit before auth from <192.168.1.193:61455>: (user 'root', 3 fails): Exited normally
> Sep 06 20:49:56 DietPi dropbear[1337]: [1337] Sep 06 20:49:56 Child connection from 192.168.1.193:61466
> Sep 06 20:49:56 DietPi dropbear[1337]: [1337] Sep 06 20:49:56 Failed loading /etc/dropbear/dropbear_dss_host_key
> Sep 06 20:50:04 DietPi dropbear[1337]: [1337] Sep 06 20:50:04 Password auth succeeded for 'xxxxxx' from 192.168.1.193:61466
> Sep 06 20:50:39 DietPi sudo[1648]:  xxxxxx : TTY=pts/0 ; PWD=/home/xxxxxx ; USER=root ; COMMAND=/usr/bin/apt upgrade
> Sep 06 20:50:39 DietPi sudo[1648]: pam_unix(sudo:session): session opened for user root(uid=0) by xxxxxx(uid=1001)

- and the contents of the config files are`cat /etc/default/dropbear`:
![image](https://github.com/user-attachments/assets/f8edfd20-c37f-4c99-bad6-6ed144ecd72b)

These don't show any issue as far as I understand. I haven't tried it, but if you would like me to reproduce the issue, I could try `/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins 0` and `/boot/dietpi/func/dietpi-set_software disable_ssh_password_logins root` again. Otherwise, I will close the issue in a few days just to be sure that things continue to work. Thanks again for your assistance.

**Reply by MichaIng** [reply]
Sorry for the late reply. If you are in mood, you can try to replicate. When it did not work at first, did you get a login prompt (at the SSH client)? Because the logs show only the 2 intended login attempts, once root failing and then another user succeeding. Just to rule out that maybe the SSH server was just not ready yet?

**Reply by centipedesquad** [reply, marked as answer]
Hey! Thanks for the reply.
I don't think I will be able to replicate in coming days. I don't have a very clear memory of it now, but I definitely got a login prompt. That is how I know I was at least reaching it :)
However, it is quite possible that I might have shared an incorrect screenshot, but I was making multiple attempts back then, apologies for that. I would propose to mark this issue as closed, as when I replicate it at a later date then I will leave a new response for the records.
